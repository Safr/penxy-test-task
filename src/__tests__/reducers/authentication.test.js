import {
  REGISTER_SUCCESS, LOGIN_SUCCESS, NOT_AUTHED, UPDATE,
} from '../../actionTypes';
import authenticationReducer from '../../reducers/authentication';

const initialState = {
  name: '',
  email: '',
  token: '',
};

describe('Authentication reducer', () => {
  const data = { name: 'name', email: 'email', access_token: 'token' };
  test('should setup default state', () => {
    const state = authenticationReducer(undefined, { type: '@@INIT' });
    expect(state).toMatchSnapshot();
  });
  test('should set register data', () => {
    const action = { type: REGISTER_SUCCESS, ...data };
    const state = authenticationReducer(initialState, action);
    expect(state).toMatchSnapshot();
  });
  test('should set login data', () => {
    const action = { type: LOGIN_SUCCESS, ...data };
    const state = authenticationReducer(initialState, action);
    expect(state).toMatchSnapshot();
  });
  test('should set updated data', () => {
    const action = { type: UPDATE, name: 'name', email: 'email' };
    const state = authenticationReducer(initialState, action);
    expect(state).toMatchSnapshot();
  });
  test('should back to initial state', () => {
    const action = { type: NOT_AUTHED };
    const state = authenticationReducer(initialState, action);
    expect(state).toMatchSnapshot();
  });
});
