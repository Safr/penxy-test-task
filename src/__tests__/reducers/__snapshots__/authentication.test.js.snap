// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication reducer should back to initial state 1`] = `
Object {
  "email": "",
  "name": "",
  "token": "",
}
`;

exports[`Authentication reducer should set login data 1`] = `
Object {
  "email": "email",
  "name": "name",
  "token": "token",
}
`;

exports[`Authentication reducer should set register data 1`] = `
Object {
  "email": "email",
  "name": "name",
  "token": "token",
}
`;

exports[`Authentication reducer should set updated data 1`] = `
Object {
  "email": "email",
  "name": "name",
  "token": "",
}
`;

exports[`Authentication reducer should setup default state 1`] = `
Object {
  "email": "",
  "name": "",
  "token": "",
}
`;
